src/ripple/nodestore/impl/Shard.h

```diff
@@ -179,43 +187,52 @@ class Shard
     // Used as an optimization for visitDifferences
     std::shared_ptr<Ledger const> lastStored_;
 
-    // Validate this ledger by walking its SHAMaps
-    // and verifying each merkle tree
+    // Marks shard immutable
+    // Lock over mutex_ required
     bool
-    valLedger(std::shared_ptr<Ledger const> const& ledger,
-        std::shared_ptr<Ledger const> const& next);
-
-    // Fetches from the backend and will log
-    // errors based on status codes
-    std::shared_ptr<NodeObject>
-    valFetch(uint256 const& hash);
-
-    // Marks shard immutable, having stored all of its ledgers
-    void
-    setComplete();
+    setComplete(std::lock_guard<std::mutex> const& lock);
 
     // Set the backend cache
+    // Lock over mutex_ required
     void
-    setCache();
+    setCache(std::lock_guard<std::mutex> const& lock);
 
     // Open/Create SQLite databases
+    // Lock over mutex_ required
     bool
-    initSQLite();
+    initSQLite(std::lock_guard<std::mutex> const& lock);
 
-    // Create SQLite entries for a ledger stored in this shard's backend
+    // Write SQLite entries for a ledger stored in this shard's backend
+    // Lock over mutex_ required
     bool
-    setSQLiteStored(std::shared_ptr<Ledger const> const& ledger);
+    setSQLiteStored(
+        std::shared_ptr<Ledger const> const& ledger,
+        std::lock_guard<std::mutex> const& lock);
 
     // Set storage and file descriptor usage stats
+    // Lock over mutex_ required
     bool
-    setFileStats();
+    setFileStats(std::lock_guard<std::mutex> const& lock);
 
     // Save the control file for an incomplete shard
+    // Lock over mutex_ required
     bool
-    saveControl();
+    saveControl(std::lock_guard<std::mutex> const& lock);
+
+    // Validate this ledger by walking its SHAMaps
+    // and verifying each merkle tree
+    bool
+    valLedger(
```

Comment:
Could this be `valLedger(ledger, next) const`?

Reply:
yes